src/lib/hooks/useTimeAgo/test.ts

```diff
@@ -0,0 +1,37 @@
+/* eslint-disable @typescript-eslint/ban-ts-comment */
+import { act } from '@testing-library/react';
+import { renderHook } from '@testing-library/react-hooks';
+
+import { useTimeAgo } from '.';
+
+describe('useTimeAgo', () => {
+    beforeEach(() => {
+        jest.useFakeTimers();
+    });
+
+    afterEach(() => {
+        jest.useRealTimers();
+        jest.clearAllTimers();
+    });
+
+    test('it formats a date', () => {
+        // Past date
+        const date = new Date();
+        const { result } = renderHook(() => useTimeAgo(date));
+        expect(result.current).toBe('just now');
+        // Future date
+        const futureDate = new Date(date.getTime() + 125000); // 2.5 minutes from now
+        const { result: resultFuture } = renderHook(() => useTimeAgo(futureDate));
+        expect(resultFuture.current).toBe('in 2 minutes');
+    });
+
+    test('when realtime is true, it automatically updates the date', () => {
+        const date = new Date(new Date().getTime() - 50000); // 50 seconds ago
+        const { result } = renderHook(() => useTimeAgo(date, true));
+        expect(result.current).toBe('50 seconds ago');
+        act(() => {
+            jest.runOnlyPendingTimers(); // fast-forward timeout in the hook
+        });
+        expect(result.current).toBe('30 minutes ago');
```

Comment:
Confused about how this jumps from 50 seconds ago to 30 minutes ago without something more explicit. Maybe there should be a test for each of the different intervals, like < 1 minute, < 1 hr, less < 24 hr and > 24 hr to make this all more clear and verify that those all do what's expected.